keps/sig-cloud-provider/20180731-cloud-provider-docs.md

```diff
@@ -0,0 +1,155 @@
+
+```
+cloud-provider-foobar/
+├── ...
+├── ...
+├── docs
+│   └── example-manifests
+│       └── in-tree/
+│           ├── apiserver.manifest                 # an example manifest of apiserver using the in-tree integation of this cloud provider
+│           ├── kube-controller-manager.manifest   # an example manifest of kube-controller-manager using the in-tree integration of this cloud provider
+│           ├── kubelet.manifest                   # an example manifest of kubelet using the in-tree integration of this cloud provider
+│       └── out-of-tree/
+│           ├── apiserver.manifest                 # an example manifest of apiserver using the out-of-tree integration of this cloud provider
+│           ├── kube-controller-manager.manifest   # an example manifest of kube-controller-manager using the out-of-tree integration of this cloud provider
+│           ├── cloud-controller-manager.manifest  # an example manifest of cloud-controller-manager using the out-of-tree integration of this cloud provider
+│           ├── kubelet.manifest                   # an example manifest of kubelet using out-of-tree integration of this cloud provider
+```
+
+#### Requirement 2: Resource Management
+
+List the latest annotations and labels that are cloud-provider dependent and can be used by the Kubernetes administrator. The contents of these files should be kept up-to-date as annotations/labels are deprecated/removed/added/updated. Labels and annotations should be grouped based on the resource they are applied to. For example the label `beta.kubernetes.io/instance-type` is applied to nodes so it should be added to `k8s.io/cloud-provider-foobar/docs/node/labels.md`.
+
+```
+cloud-provider-foobar/
+├── ...
+├── ...
+├── docs
+│   └── resources
+│       └── node/
+│           ├── labels.md        # outlines what annotations that can be used on a Node resource
+│           ├── annotations.md   # outlines what annotations that can be used on a Node resource
+│           ├── README.md        # outlines any other cloud provider specific details worth mentioning regarding Nodes
+│       └── service/
+│           ├── labels.md        # outlines what annotations that can be used on a Service resource
+│           ├── annotations.md   # outlines what annotations that can be used on a Service resource
+│           ├── README.md        # outlines any other cloud provider specific details worth mentioning regarding Services
+│       └── persistentvolumes/
+│           ├── ...
+│       └── ...
+│       └── ...
+```
+
+#### Requirement 3: Kubeadm Setup Guide
+
+Add a setup guide on how to configure your cloud provider with kubeadm. The setup guide may reference another blog post or tutorial that covers how to do this.
```

Comment:
I don't think we should require cloud provider setup to be done with kubeadm.

Reply:
Makes sense, I'll remove this.